### ISSUE TITLE
[ROADMAP] A customer logged when he visit the register page must to be redirected

### ISSUE THREAD

**Issue by Chrysweel** [Sylius/Sylius#9984]
A customer logged in Sylius, when he visit register page must to be redirected home.
If customer can enter in register page, he will try register again... 

The page to redirect the user should be configurable by the ecommerce.

**Comment by CoderMaggie** [comment]
The register link is not available via fronted. They can access the register page only by going to the /register url. I do not see any particular mistake in allowing the logged in user to register again, without logout. Though if they provide the same email they have already registered with, it will be impossible to finish the registration.

If you need to change that behaviour, you can customize it :)

**Comment by bambamboole** [comment]
I think something like redirecting on the customer account and add a notification that he ist logged in would be a nice behaviour.

**Comment by Chrysweel** [comment]
Yes @CoderMaggie the user can visit the registration page, directly through google, through a link in an email, or through his own browser history.

If a user who is already logged in, visits the registration page, several things can happen:

A) You can register again, creating a duplicate account in the ecommerce for the same user, and when you go to see your purchase history you will see it empty. And the user will not understand anything.

B) Try to register with your email and give an error, the user can not understand what happens and end up registering with another account, or leaving the website.

Honestly, I can not think of a good possibility, when a user with logged visits the registration page ....

**Comment by alekseyp** [comment]
I would agree with everyone that -  If user is logged in and user navigates to registration page - user should be redirected into user account page automatically.

**Comment by Chrysweel** [comment]
Do we have any conclusions about it?

**Comment by alekseyp** [comment]
Ohno:(

**Comment by CoderMaggie** [comment]
As I have stated last year, I still believe there is no particular harm in the current behaviour, many shops do that. Do not worry, the user who is already registered will not be able to register with the same email again and if they try they will be informed that there is already an account with this email.

If you want I can put this rework on the roadmap and we can see how many need this improvement. @Chrysweel do you mind if I edit the description and title to be clearer? :)

**Comment by gabiudrescu** [comment]
one tiny suggestion: I would replace User with Logged in user in the description of this task. 

otherwise, yes, I totally vote for it, as it looks like a good UX feature.

**Comment by alekseyp** [comment]
.

**Comment by gabiudrescu** [comment]
Judging by the number of positive votes this issue received (9 + 5 + 5), I think it worth having a clear solution: 
 - accept (and implement) the suggestion
 - reject (no hard feelings) and close the issue to avoid future debates over this


my 2 cents is that this feature should be implemented, as a logged in user should not be able to reach register page or login page, as this makes one think there's been an error and they need to login / register again, creating an infinite loop (of frustration). 

we had this feedback from our (internal) users with 0.19 version of Sylius, where the admin login was not always redirecting to dashboard, making users think they either forgot their password or the system is broken. 

increasing the number of steps and the friction between the platform and the user is troublesome and it increases the negative feeling towards the system.

**Comment by alekseyp** [comment]
.

**Comment by maximehuran** [comment]
Same behaviour on Sylius 1.10 and 1.11-DEV

**Comment by CoderMaggie** [comment]
@maximehuran intended, we haven't been changing anything yet 🖖🏼 

Let's revisit this topic then, I will bring it to our internal attention at Sylius during 1.11 development.

**Comment by Chrysweel** [comment]
Something new about this?

I have a controller override in our project from 29 november of 2018 with a comment.

` * Overwrite this controller to redirect user logged when visit page register, see issue https://github.com/Sylius/Sylius/issues/9984 when Sylius do it by default, we can remove this custom code`

**Comment by sonvir249** [comment]
is this still valid?
